### PR TITLE
Changed DataTable to add select and onSelect properties

### PR DESCRIPTION
**Pull request by ericsoderberghp**
#### What does this PR do?

Changed DataTable to add select and onSelect properties

#### Where should the reviewer start?

doc.js

#### What testing has been done on this PR?

added storybook story and unit tests

#### How should this be manually tested?

storybook

#### What are the relevant issues?

#4444

#### Do the grommet docs need to be updated?

automatic

#### Should this PR be mentioned in the release notes?

yes

#### Is this change backwards compatible or is it a breaking change?

backwards compatible

### CODE REVIEW

File: src/js/components/DataTable/Header.js [hunk 1]
```diff
@@ -51,6 +58,24 @@ const Header = ({
...
 
+        {(selected || onSelect) && (
+          <TableCell>
+            <CheckBox
+              checked={selected.length === data.length}
+              indeterminate={
+                selected.length > 0 && selected.length < data.length
+              }
+              onChange={() => {
+                if (selected.length > 0) onSelect([]);
```
Comment (ShimiSun) [inline src/js/components/DataTable/Header.js:69]: Not sure I understand the logic here, can you elaborate?

Comment (ericsoderberghp) [reply src/js/components/DataTable/Header.js:70]: I've added some comments in the code

File: src/js/components/DataTable/doc.js [hunk 2]
```diff
@@ -233,6 +241,13 @@ export const doc = DataTable => {
       where the background, border, and pad accept the same values as
       the same named properties on DataTable.`,
     ),
+    select: PropTypes.arrayOf(
+      PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+    ).description(
+      `Which rows should be selected. The values in this array should match
+      the primaryKey or columns[].primary keyed value for the row's data
+      object.`,
```
Comment (ShimiSun) [inline src/js/components/DataTable/doc.js:249]: Should we add something about the control of the selected rows by using `onSelect`?

File: src/js/components/DataTable/doc.js [hunk 3]
```diff
@@ -185,6 +185,14 @@ export const doc = DataTable => {
       names and values which are the search text strings. This is typically
       employed so a back-end can be used to search through the data.`,
     ),
+    onSelect: PropTypes.func.description(
```
Comment (ShimiSun) [inline src/js/components/DataTable/doc.js:188]: Using `onSelect` without the `select` property makes any sense?

**Review by ShimiSun — COMMENTED** [review]
There is an inconsistent behavior with how we demonstrate the[ CheckBox indeterminate state,](https://deploy-preview-4483--sad-tereshkova-173d07.netlify.app/?path=/story/checkbox--indeterminate) and DataTable Select story. when the indeterminate checkbox is on the indeterminate mode, clicking on the indeterminate checkbox selects all options, while on DataTable it unselects all options. The Indeterminate (story) logic makes more sense to me than how the DataTable operates.

**Comment by ericsoderberghp** [comment]
> There is an inconsistent behavior with how we demonstrate the[ CheckBox indeterminate state,](https://deploy-preview-4483--sad-tereshkova-173d07.netlify.app/?path=/story/checkbox--indeterminate) and DataTable Select story. when the indeterminate checkbox is on the indeterminate mode, clicking on the indeterminate checkbox selects all options, while on DataTable it unselects all options. The Indeterminate (story) logic makes more sense to me than how the DataTable operates.

Fixed

File: src/js/components/DataTable/doc.js (hunk 2 above)
Comment (ericsoderberghp) [reply src/js/components/DataTable/doc.js:249]: added

File: src/js/components/DataTable/doc.js (hunk 3 above)
Comment (ericsoderberghp) [reply src/js/components/DataTable/doc.js:189]: Allowed using it uncontrolled in this way.

**Pull request merged**